eZ/Publish/API/Repository/Tests/SearchServiceAggregationTest.php

```diff
@@ -0,0 +1,886 @@
+use eZ\Publish\API\Repository\Values\Content\Query\Aggregation\RawTermAggregation;
+use eZ\Publish\API\Repository\Values\Content\Query\Aggregation\SectionTermAggregation;
+use eZ\Publish\API\Repository\Values\Content\Query\Aggregation\UserMetadataTermAggregation;
+use eZ\Publish\API\Repository\Values\Content\Query\Aggregation\VisibilityTermAggregation;
+use eZ\Publish\API\Repository\Values\Content\Query\AggregationInterface;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion\ContentTypeIdentifier;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion\MatchAll;
+use eZ\Publish\API\Repository\Values\Content\Search\AggregationResult;
+use eZ\Publish\API\Repository\Values\Content\Search\AggregationResult\RangeAggregationResult;
+use eZ\Publish\API\Repository\Values\Content\Search\AggregationResult\RangeAggregationResultEntry;
+use eZ\Publish\API\Repository\Values\Content\Search\AggregationResult\StatsAggregationResult;
+use eZ\Publish\API\Repository\Values\Content\Search\AggregationResult\TermAggregationResult;
+use eZ\Publish\API\Repository\Values\Content\Search\AggregationResult\TermAggregationResultEntry;
+use eZ\Publish\API\Repository\Values\ContentType\ContentType;
+use eZ\Publish\API\Repository\Values\ContentType\FieldDefinitionCreateStruct;
+use eZ\Publish\API\Repository\Values\ObjectState\ObjectState;
+use eZ\Publish\Core\FieldType\Checkbox\Value as CheckboxValue;
+use eZ\Publish\Core\FieldType\Time\Value as TimeValue;
+
+/**
+ * Test case for aggregations in the SearchService.
+ *
+ * @see \eZ\Publish\API\Repository\SearchService
+ * @group integration
+ * @group search
+ * @group aggregations
+ */
+final class SearchServiceAggregationTest extends BaseTest
+{
+    private const EXAMPLE_COUNTRY_FIELD_VALUES = [
+        ['PL', 'US'],
+        ['FR', 'US'],
+        ['US'],
+        ['GA', 'PL', 'FR'],
+        ['FR', 'BE', 'US'],
+    ];
+
+    private const EXAMPLE_KEYWORD_FIELD_VALUES = [
+        ['foo'],
+        ['foo', 'bar'],
+        ['foo', 'bar', 'baz'],
+    ];
+
+    private const EXAMPLE_SELECTION_FIELD_VALUES = [
+        [0],
+        [0, 1],
+        [0, 1, 2],
+    ];
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $searchService = $this->getRepository()->getSearchService();
+        if (!$searchService->supports(SearchService::CAPABILITY_AGGREGATIONS)) {
+            $this->markTestSkipped("Search engine doesn't support aggregations");
+        }
+    }
+
+    /**
+     * @dataProvider dataProviderForTestAggregation
+     */
+    public function testFindContentWithAggregation(
+        AggregationInterface $aggregation,
+        AggregationResult $expectedResult
+    ): void {
+        $searchService = $this->getRepository()->getSearchService();
+
+        $query = new Query();
+        $query->aggregations[] = $aggregation;
+        $query->filter = new MatchAll();
+        $query->limit = 0;
+
+        $this->assertEquals(
```

Comment:
```suggestion
        self::assertEquals(
```
similar change applies to all of the new code.

Reply:
Fixed in 672a0ea